pkg/controller/nodetune/tune.go

```diff
@@ -110,37 +70,30 @@ func scyllaDataDirMountHostPaths(ctx context.Context, criClient cri.Client, scyl
 	return dataDirs.List(), nil
 }
 
-func cpusetFromCRI(ctx context.Context, client cri.Client, cid string) (cpuset.CPUSet, error) {
-	cs, err := client.Inspect(ctx, cid)
+func cpusetFromKubelet(ctx context.Context, podResourcesClient kubelet.PodResourcesClient, scyllaPod *corev1.Pod, containerName string) (cpuset.CPUSet, error) {
+	prs, err := podResourcesClient.List(ctx)
 	if err != nil {
-		return cpuset.CPUSet{}, fmt.Errorf("can't inspect container %q, %w", cid, err)
-	}
-
-	if cs.Info.RuntimeSpec == nil {
-		klog.V(2).InfoS("No container status available", "ContainerID", cid)
-		return cpuset.CPUSet{}, nil
+		return cpuset.CPUSet{}, fmt.Errorf("can't list pod resources: %w", err)
 	}
 
-	containerCpuSet, err := cpuset.Parse(cs.Info.RuntimeSpec.Linux.Resources.CPU.Cpus)
-	if err != nil {
-		return cpuset.CPUSet{}, fmt.Errorf("can't parse container %q cpuset %q, %w", cid, cs.Info.RuntimeSpec.Linux.Resources.CPU.Cpus, err)
+	for _, pr := range prs {
+		if pr.Namespace == scyllaPod.Namespace && pr.Name == scyllaPod.Name {
+			for _, cr := range pr.Containers {
+				if cr.Name == containerName {
```

Comment:
FindContainerWithName takes `[]corev1.Container` not `[]v1.ContainerResources`. Helper would be good if there would be another function using it, but since we only have one, and I don't see use case for another one I left it inlined.
I refactored the code to reduce indentation required though.

Reply:
my bad about the existing method. both external method and restructuring the nesting works for me, thx